File: hosted/catalog/names_test.go
package catalog

import (
	"slices"
	"testing"
)

func TestDisplayName(t *testing.T) {
	for in, want := range map[string]string{
		"cubeship-umami-template":       "Umami",
		"cubeship-uptime-kuma-template": "Uptime Kuma",
		"Cubeship-N8N-Template":         "N8N",
		"plausible":                     "Plausible",
		"my_app":                        "My App",
		"cubeship-template":             "Template",
	} {
		if got := DisplayName(in); got != want {
			t.Errorf("DisplayName(%q) = %q, want %q", in, got, want)
		}
	}
}

func TestTagsLeaveTheListingTopicOut(t *testing.T) {
	if got := tagsOf([]string{Topic, "analytics"}, ""); !slices.Equal(got, []string{"analytics"}) {
		t.Errorf("tags = %v", got)
	}
}

File: hosted/site/src/lib/templates/names.test.ts
import { describe, expect, it } from "vitest";
import { displayName, tagsOf } from "./names";

describe("displayName", () => {
  it("reads the app out of the naming convention", () => {
    expect(displayName("cubeship-umami-template")).toBe("Umami");
    expect(displayName("cubeship-uptime-kuma-template")).toBe("Uptime Kuma");
  });

  it("leaves a name outside the convention readable", () => {
    expect(displayName("plausible")).toBe("Plausible");
    expect(displayName("my_app")).toBe("My App");
  });
});

describe("tagsOf", () => {
  it("drops the topic every template carries", () => {
    expect(tagsOf(["cubeship-template", "analytics"])).toEqual(["analytics"]);
  });
});
